### PR TITLE
feat(docker): setup infra for frontend under development

### PR DESCRIPTION
**Pull request by lelouvincx**
## Description

Setup docker compose for frontend for development team.

## How Has This Been Tested?

Checkout the branch and try:
```bash
make build
make up
```
Then try editing some code from `chatbot-core/frontend` to see change effects.

## Related Issue(s)

Issue #35 

## Checklist:

- [x] All of the automated tests pass
- [ ] All PR comments are addressed and marked resolved
- [x] If there are migrations, they have been rebased to latest main
- [x] If there are new dependencies, they are added to the requirements
- [x] If there are new environment variables, they are added to all of the deployment methods
- [x] If there are new APIs that don't require auth, they are added to PUBLIC_ENDPOINT_SPECS
- [x] Docker images build and basic functionalities work
- [ ] Author has done a final read through of the PR right before merge

### CODE REVIEW

**Comment by lekiet1214** [comment]
@lelouvincx i have added configuration for webserver and nginx, please check and open for review

**Comment by lelouvincx** [comment]
> @lelouvincx i have added configuration for webserver and nginx, please check and open for review

I'm on it

**Review by lelouvincx — COMMENTED** [review]
@lekiet1214 I encountered this error after pulling and building frontend image.
image

**Comment by lekiet1214** [comment]
@lelouvincx i have pushed the fix, please rv

File: deployment/docker_compose/docker-compose.dev.yaml [hunk 1]
```diff
@@ -110,6 +110,40 @@ services:
         max-size: "50m"
         max-file: "6"
 
+  web_server:
```
Comment (tuantran0910) [inline deployment/docker_compose/docker-compose.dev.yaml:113]: Add `network` field.

File: deployment/docker_compose/docker-compose.dev.yaml [hunk 2]
```diff
@@ -110,6 +110,40 @@ services:
...
+      dockerfile: Dockerfile
+    depends_on:
+      - api_server
+    restart: on-failure
+    container_name: web_server
+    hostname: web_server
+    ports:
+      - "3000:3000"
+
+  nginx:
```
Comment (tuantran0910) [inline deployment/docker_compose/docker-compose.dev.yaml:125]: Add `network` field.

**Comment by lekiet1214** [comment]
I have investigated this issue further.
The `package-lock.json` was in `.gitignore`, thus missing from the source.

**Comment by lelouvincx** [comment]
Waiting for frontend to be integrated with backend...

**Comment by lekiet1214** [comment]
Will merge with #126

**Pull request closed without merging**